CONTRIBUTING.md

```diff
@@ -491,19 +491,27 @@ To get a sense for what changes are considered mass rebuilds, see [previously me
 ## Commit conventions
 [commit-conventions]: #commit-conventions
 
-- Create a commit for each logical unit.
+- Commits should be independent and understandable by themselves.
 
-- Check for unnecessary whitespace with `git diff --check` before committing.
+  For example:
+  - For example, when adding yourself as maintainer in the same pull request,
+    make a separate commit with the message `maintainers: add <handle>`.
+    Add the commit before those making changes to the package or module.
+    See [Nixpkgs Maintainers](./maintainers/README.md) for details.
 
-- If you have commits `pkg-name: oh, forgot to insert whitespace`: squash commits in this case. Use `git rebase -i`.
-  See [Squashing Commits](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#_squashing) for additional information.
+- Later commits that undo changes from earlier commits should be combined.
```

Comment:
I guess one exception would be commits that are meant to be reverted later ([example](https://github.com/NixOS/nixpkgs/pull/67576/commits/bc17b33471454e8bce6edf92de8030a121c15dbd)). Not sure if it makes sense to mention that case.

Reply:
```suggestion
- Later commits that undo changes from earlier commits should usually be combined.
```

I think adding "usually" implies that this is a rule-of-thumb rather than an actual rule. That should be enough for those kinda scenarios?